staging/src/k8s.io/component-base/featuregate/feature_gate.go

```diff
@@ -423,6 +423,37 @@ func (f *featureGate) AddVersioned(features map[Feature]VersionedSpecs) error {
 		return fmt.Errorf("cannot add a feature gate after adding it to the flag set")
 	}
 
+	// Validate new specs are well-formed
+	for name, specs := range features {
+		var lastVersion *version.Version
+		var wasBeta, wasGA bool
+		for i, spec := range specs {
+			if spec.Version == nil {
+				return fmt.Errorf("feature %q did not provide a version", name)
+			}
+			// gates that begin as deprecated must indicate their prior state
+			if i == 0 && spec.PreRelease == Deprecated && spec.Version.Minor() != 0 {
+				return fmt.Errorf("feature %q introduced as deprecated must provide a 1.0 entry indicating initial state", name)
+			}
+			if i > 0 {
+				// versions must strictly increase
+				if !lastVersion.LessThan(spec.Version) {
+					return fmt.Errorf("feature %q lists version transitions in non-increasing order (%s >= %s)", name, spec.Version, lastVersion)
```

Comment:
lastVersion and spec.Version should be swapped right? In this case `lastVersion >= spec.Version` or `spec.Version < lastVersion`